pkg/target/icd.go

```diff
@@ -6,61 +6,60 @@ import (
 )
 
 func (t *Target) WillApplyEle(tag combat.ICDTag, grp combat.ICDGroup, char int) float64 {
-
-	//no icd if no tag
+	// no icd if no tag
 	if tag == combat.ICDTagNone {
 		return 1
 	}
 
-	//check if we need to start timer
+	// check if we need to start timer
 	x := t.icdTagOnTimer[char][tag]
 	if !t.icdTagOnTimer[char][tag] {
 		t.icdTagOnTimer[char][tag] = true
 		t.ResetTagCounterAfterDelay(tag, grp, char)
 	}
 
 	val := t.icdTagCounter[char][tag]
-	//increment the counter
-	//if counter > length, then reset back to 0
 	t.icdTagCounter[char][tag]++
-	if t.icdTagCounter[char][tag] == len(combat.ICDGroupEleApplicationSequence[grp]) {
-		t.icdTagCounter[char][tag] = 0
+
+	// if counter > length, then use 0 for group seq
+	groupSeq := 0.0
+	if val < len(combat.ICDGroupEleApplicationSequence) {
```

Comment:
Shouldn't this be` if val < len(combat.ICDGroupEleApplicationSequence[grp])? (missing [grp])